src/common/metrics.rs

```diff
@@ -180,8 +184,13 @@ impl MetricsProvider for AppFeaturesTelemetry {
     }
 }
 
-impl MetricsProvider for CollectionsTelemetry {
-    fn add_metrics(&self, metrics: &mut MetricsData, prefix: Option<&str>) {
+impl CollectionsTelemetry {
```

Comment:
Since we need access to the peer id of the current instance, I had to change this from a trait to a type function. We don't use this trait for it's trait benefits, so this approach seemed best to me.
If there is a different preferred approach, please let me know.